materials/section-2/README.md

```diff
@@ -0,0 +1,149 @@
+# Backend Programming
+
+In this section we will delve into backend programming with Ring and Compojure. Compojure is a lightweight routing library built on top of Ring which is an http library for Clojure backend development.
+
+Breakdown:
+
+- Ring -> Http server
+- Compojure -> Routing lib
+
+To scaffold a Compojure project, run `lein new compojure <name>` (replace name with a the name of your project). This command will create a Compojure project with the following folder structure:
+
+```
+<name>/
+├── README.md
+├── project.clj
+├── resources
+│   └── public
+├── src
+│   └── <name>
+│       └── handler.clj
+└── test
+    └── <name>
+        └── handler_test.clj
+```
+
+The most interesting files are without a doubt, `project.clj` and `src/<name>/handler.clj`. The `project.clj` file is the file that describes the project to Leiningen. The project dependencies, name, description etc. are describes inside the `project.clj` file. Incase of a Compojure project, it also defines the entry point for the web-server: `:ring {:handler <name>.handler/app}`. This tells Ring where to look for the entrypoint so it can bootstrap the server. The second interesting file, is the `handler.clj` file. This is where the application is bootstrapped. It will look something like this:
```

Comment:
"The project dependencies, name, description etc. are **described** inside ..."

"**In the case of a Compojure project**, it ..."